### ISSUE TITLE
Exception when trying to Get Watcher with Weekly Schedule Trigger 

### ISSUE THREAD

**Issue by Gianluca-M** [elastic/elasticsearch-net#5725]
**NEST/Elasticsearch.Net version**: 7.12.1 (Edit: Also in 7.13.0)

**Elasticsearch version**: 7.12.1

**Description of the problem including expected versus actual behavior**:
I have created a watcher using Nest `client.Watcher.Put(...)` with a Weekly schedule trigger, which I created using a `ScheduleDesriptor`:
```
scheduleDescriptor.Weekly(ws => ws.Add(w => w.On(Day.Friday).At("17:00")));
```
 The created Watcher Trigger Config is as follows: (using `GET _watcher/watch/watcherID` in Elasticsearch)
```
{
  "trigger": {
    "schedule": {
      "weekly": [
        {
          "on": [
            "FRI"
          ],
          "at": [
            "17:00"
          ]
        }
      ]
    }
  }
}
```
When I now try to get the Watcher using Nest `client.Watcher.Get(watcherId)` it throws an Exception:
```
Elasticsearch.Net.UnexpectedElasticsearchClientException: Requested value 'FRI' was not found. ---> System.ArgumentException: Requested value 'FRI' was not found.
   at System.Enum.TryParseEnum(Type enumType, String value, Boolean ignoreCase, EnumResult& parseResult)
   at System.Enum.Parse(Type enumType, String value, Boolean ignoreCase)
   at Elasticsearch.Net.Utf8Json.Formatters.EnumFormatter`1.Deserialize(JsonReader& reader, IJsonFormatterResolver formatterResolver)
   at Elasticsearch.Net.Utf8Json.Formatters.CollectionFormatterBase`4.Deserialize(JsonReader& reader, IJsonFormatterResolver formatterResolver)
   at Deserialize(Object[] , JsonReader& , IJsonFormatterResolver )
   at Nest.ReadAsFormatter`2.Deserialize(JsonReader& reader, IJsonFormatterResolver formatterResolver)
   at Elasticsearch.Net.Utf8Json.Formatters.CollectionFormatterBase`4.Deserialize(JsonReader& reader, IJsonFormatterResolver formatterResolver)
   at Nest.ScheduleFormatter`3.Deserialize(JsonReader& reader, IJsonFormatterResolver formatterResolver)
   at Deserialize(Object[] , JsonReader& , IJsonFormatterResolver )
   at Nest.ReadAsFormatter`2.Deserialize(JsonReader& reader, IJsonFormatterResolver formatterResolver)
   at Deserialize(Object[] , JsonReader& , IJsonFormatterResolver )
   at Deserialize(Object[] , JsonReader& , IJsonFormatterResolver )
   at Nest.ReadAsFormatter`2.Deserialize(JsonReader& reader, IJsonFormatterResolver formatterResolver)
   at Deserialize(Object[] , JsonReader& , IJsonFormatterResolver )
   at Elasticsearch.Net.Utf8Json.JsonSerializer.Deserialize[T](Byte[] bytes, Int32 offset, IJsonFormatterResolver resolver)
   at Elasticsearch.Net.Utf8Json.JsonSerializer.Deserialize[T](Stream stream, IJsonFormatterResolver resolver)
   at Elasticsearch.Net.DiagnosticsSerializerProxy.Deserialize[T](Stream stream)
   at Elasticsearch.Net.ResponseBuilder.SetBody[TResponse](ApiCallDetails details, RequestData requestData, Stream responseStream, String mimeType)
   at Elasticsearch.Net.ResponseBuilder.ToResponse[TResponse](RequestData requestData, Exception ex, Nullable`1 statusCode, IEnumerable`1 warnings, Stream responseStream, String mimeType)
   at Elasticsearch.Net.HttpWebRequestConnection.Request[TResponse](RequestData requestData)
   at Elasticsearch.Net.RequestPipeline.CallElasticsearch[TResponse](RequestData requestData)
   at Elasticsearch.Net.Transport`1.Request[TResponse](HttpMethod method, String path, PostData data, IRequestParameters requestParameters)
   --- End of inner exception stack trace ---
   at Elasticsearch.Net.Transport`1.Request[TResponse](HttpMethod method, String path, PostData data, IRequestParameters requestParameters)
   at Nest.Specification.WatcherApi.WatcherNamespace.Get(Id id, Func`2 selector)
```
Note that the watcher works as intended, I can see in Kibana that it correctly triggered.

When I use a Watcher with any other Trigger Schedule (Hourly, Daily, Monthly, Yearly, Cron, Interval) it works without any problems and correctly returns the watcher.

**Steps to reproduce**:
 1. Create a watcher with a Weekly Schedule
 2. Try to get the watcher using `client.Watcher.Get(...)`

**Expected behavior**
It should return a valid `GetWatchResponse`

**Another question**
I also have a question regarding cron expression schedules: 
How can I create a watcher with multiple cron expression schedules in NEST? Normally in Elasticsearch, I would simply do:
```
"trigger": {
    "schedule": {
      "cron" : [
        "0 0/2 * ? * MON-FRI",
        "0 1-59/2 * ? * SAT-SUN"
      ]
    }
  }
```
But in NEST, since the Cron ScheduleDescriptor does not allow a List of Cron Expressions as argument, I tried to do the following:
`scheduleDescriptor.Cron("0 0/2 * ? * MON-FRI").Cron("0 1-59/2 * ? * SAT-SUN")`
However, in this case, only the last cron expression will actually be saved in the watcher, i.e. the resulting watcher schedule would be the following:
```
"trigger": {
    "schedule": {
      "cron" : "0 1-59/2 * ? * SAT-SUN"
    }
  }
```
Is there any way to define multiple cron expression schedules for a watcher?

**Comment by stevejgordon** [comment]
Hi @Gianluca-M. Thanks for reporting this. I can see the issue which is that the server is returning the day as `FRI` which our enum does not contain. I'll review this further and look to see how we can support this.

To your second question, currently, the client only supports a single cron value. This one may be harder to resolve without making a breaking change to our model for the schedule. Again, this is something I can take away and review.

In the meantime, you can work around this by dropping down to the low-level client, although you will then need to serialise/deserialise the request and responses manually, which I appreciate is not ideal.

**Comment by Gianluca-M** [comment]
HI @stevejgordon 
Thanks for the update and work on this issue.
While working more with watchers, I also noticed a bug with the Yearly schedule: No matter which month I select in the yearly schedule, the resulting watcher always has "JAN" (January) as the month.
e.g. I add the following yearly schedule: `s.Yearly(ys => ys.Add(y => y.In(Month.July).On(12).At("08:00")))`
The expected watcher should have July as its month but the actual created watcher has the following schedule:
```
"yearly": [
  {
    in": [
      "JAN"
    ],
    "on": [
      12
    ],
    "at": [
      "08:00"
    ]
  }
]
```
Should I open a new issue about this bug, or is it ok if I just leave it here?

**Comment by stevejgordon** [comment]
@Gianluca-M Thanks for raising these bugs. No need for an extra issue in this case. I've been working on this today and I think I have a solution soon for the array of cron expressions support nearly good to submit. I've been adding more tests around the trigger schedule, so I can include one for the yearly schedule and hopefully identify what's going wrong there too.

**Comment by stevejgordon** [comment]
@Gianluca-M I have merged some fixes around schedule triggers which should address the issues you've identified above. This will be released as part of 7.14.0.
